src/Microsoft.VisualStudio.LanguageServices.Razor/RazorDiagnosticJsonConverter.cs

```diff
@@ -47,12 +46,6 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
 
                 return RazorDiagnostic.Create(descriptor, sourceSpan);
             }
-            else if (string.Equals(typeName, typeof(LegacyRazorDiagnostic).FullName, StringComparison.Ordinal))
```

Comment:
This code should be simplified further, we no longer need to deal with type names and we can just directly deserialize the `DefaultRazorDiagnostic`. I think it's OK if you do that in a follow up, but this was the issue that raised the priority of this.

Reply:
Actually a follow up is better, I think we'll want to add a microbenchmark for this and do some profiling.

Reply:
Filed https://github.com/aspnet/Razor/issues/1886 to track this.